lib/queue.rb

```diff
@@ -1,31 +1,72 @@
 class Queue
+  MAX_SIZE = 20
 
   def initialize
     # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new
+    @front = @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if (@front == 0 && @back == MAX_SIZE - 1) || (@back == (@front -1) % MAX_SIZE - 1)
+      raise Exception.new("Queue is full.")
+    elsif @store.empty? 
+      @front = 0
+      @back = 0
+    elsif @back == (MAX_SIZE - 1) && @front != 0
+      @back = 0
+    else 
+      @back += 1
+    end
```

Comment:
This could also be written as:

```suggestion
    else
      @back = (@back + 1) % MAX_SIZE
    end
```